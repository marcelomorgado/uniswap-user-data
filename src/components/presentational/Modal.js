import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import { default as MuiModal } from "@material-ui/core/Modal";

const modalStyles = theme => {
  const top = 50;
  const left = 50;

  return {
    paper: {
      position: "absolute",
      width: "80%",
      backgroundColor: theme.palette.background.paper,
      boxShadow: theme.shadows[5],
      padding: theme.spacing.unit * 4,
      outline: "none",
      top: `${top}%`,
      left: `${left}%`,
      transform: `translate(-${top}%, -${left}%)`,
    },
  };
};

function Modal({ classes, onClose, open, children }) {
  return (
    <div>
      <MuiModal
        aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
        open={open}
        onClose={onClose}
      >
        <div className={classes.paper}>{children}</div>
      </MuiModal>
    </div>
  );
}

Modal.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  children: PropTypes.object,
};

export default withStyles(modalStyles)(Modal);
