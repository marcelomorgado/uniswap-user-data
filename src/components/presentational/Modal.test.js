import React from "react";
import { shallow } from "enzyme";
import Modal from "./Modal";

it("renders the component", () => {
  const props = {
    open: true,
    onClose: () => {},
    children: {},
  };

  expect(shallow(<Modal {...props} />)).toMatchSnapshot();
});
