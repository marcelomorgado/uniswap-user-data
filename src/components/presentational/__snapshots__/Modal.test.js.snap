// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Modal
  classes={
    Object {
      "paper": "Modal-paper-1",
    }
  }
  onClose={[Function]}
  open={true}
>
  <Component />
</Modal>
`;
